src/meta.php

```diff
@@ -86,6 +86,26 @@ function register_meta_helper(
 		$meta_key
 	);
 
+	// Allow setting meta for all of an object type.
+	if (
+		(
+			is_array( $object_slugs ) &&
+			count( $object_slugs ) === 1 &&
+			'all' === $object_slugs[0]
+		) ||
+		(
+			is_string( $object_slugs ) &&
+			'all' === $object_slugs
+		)
+	) {
+		return register_meta( $object_type, $meta_key, $args );
+	}
+
+	// Fix potential errors since we're allowing `$object_slugs` to be a string or array.
+	if ( is_string( $object_slugs ) ) {
+		$object_slugs = [ $object_slugs ];
+	}
```

Comment:
This could just as easily be a thrown exception. I opted for fixing the potential type error and trusting the value of the type as if it were an empty array, but I'm not married to this approach.

Reply:
😄